gabbi/handlers/jsonhandler.py

```diff
@@ -76,29 +76,45 @@ def extract_json_path_value(data, path):
     def action(self, test, path, value=None):
         """Test json_paths against json data."""
         # Do template expansion in the left hand side.
-        path = test.replace_template(path)
+        lhs_path = test.replace_template(path)
+        rhs_path = rhs_match = None
         try:
-            match = self.extract_json_path_value(
-                test.response_data, path)
+            lhs_match = self.extract_json_path_value(
+                test.response_data, lhs_path)
         except AttributeError:
             raise AssertionError('unable to extract JSON from test results')
         except ValueError:
-            raise AssertionError('json path %s cannot match %s' %
-                                 (path, test.response_data))
+            raise AssertionError('left hand side json path %s cannot match '
+                                 '%s' % (path, test.response_data))
 
         # read data from disk if the value starts with '<@'
         if isinstance(value, str) and value.startswith('<@'):
+            # Do template expansion in the rhs if rhs_path is provided.
+            if ':' in value:
+                value, rhs_path = value.split(':')
```

Comment:
We probably need to think about how we want to manage the separator. If the filename or the jsonpath has a `:` in it, we've got some issues. Unless we want to do tokenizing (please no!) there's not going to be a foolproof way to deal with that, but there are probably some ways we can make this more robust, but I'm not sure which is the best. Some ideas:

* if we assume there's no `:` in the jsonpath we can rsplit
* if we assume there's `:` in the file we can split, 1 as you've mentioned elsewhere
* we could split on `:$` (thinking if it as a signal for "here comes some jsonpath)
* we could split on `::` or ` :: ` as mentioned in the IRC chat

I don't have a strong opinion on this, but it is worth taking the time to think up something.